build/common.sh

```diff
@@ -272,7 +272,7 @@ function kube::build::docker_image_exists() {
 
   # We cannot just specify the IMAGE here as `docker images` doesn't behave as
   # expected.  See: https://github.com/docker/docker/issues/8048
-  "${DOCKER[@]}" images | grep -Eq "^${1}\s+${2}\s+"
+  "${DOCKER[@]}" images | grep -Eq "${1}\s+${2}\s+"
```

Comment:
I'm on fedora 22, and I upgrade the docker to 1.7.1 by enable the update-testing repo.
It always pull from docker.io
```
$ docker pull nginx
latest: Pulling from docker.io/nginx
aface2a79f55: Pull complete 
5dd2638d10a1: Pull complete 
97df1ddba09e: Pull complete 
56c99fa886e8: Pull complete 
cd27805ea89f: Pull complete 
5e95ac0e9a79: Pull complete 
9f36f81a1ccb: Pull complete 
4c1809b04591: Pull complete 
98c9ccd75644: Pull complete 
6886fb5a9b8d: Already exists 
902b87aaaec9: Already exists 
9a61b6b1315e: Already exists 
docker.io/nginx:latest: The image you are pulling has been verified. Important: image verification is a tech preview feature and should not be relied on to provide security.
Digest: sha256:a2b8bef3338643176198510fec71c18ceb7834e6f7c45b039ec6e963f43b4005
Status: Downloaded newer image for docker.io/nginx:latest

$ docker version
Client version: 1.7.1.fc22
Client API version: 1.19
Package Version (client): docker-1.7.1-1.gitb6416b7.fc22.x86_64
Go version (client): go1.4.2
Git commit (client): b6416b7/1.7.1
OS/Arch (client): linux/amd64
Server version: 1.7.1.fc22
Server API version: 1.19
Package Version (server): docker-1.7.1-1.gitb6416b7.fc22.x86_64
Go version (server): go1.4.2
Git commit (server): b6416b7/1.7.1
OS/Arch (server): linux/amd64

$ docker images  nginx
REPOSITORY          TAG                 IMAGE ID            CREATED             VIRTUAL SIZE
docker.io/nginx     latest              6886fb5a9b8d        6 days ago          132.8 MB
```
And, I think it's more safe to remove the "^" in the regex, it's compatible with with prefix or without prefix.